dali/pipeline/workspace/host_workspace.h

```diff
@@ -100,7 +100,15 @@ class DLL_PUBLIC HostWorkspace : public WorkspaceBase<HostInputType, HostOutputT
     return *thread_pool_;
   }
 
+  bool has_stream() const override {
+    return false;
+  };
+
  private:
+  cudaStream_t stream_impl() const override {
```

Comment:
I would consider throwing here.
Also, as @JanuszL pointed out, the default is 0, not that it makes a lot of difference.